src/core/configuration/__tests__/tree.test.ts

```diff
@@ -0,0 +1,216 @@
+import { IConfigNode } from "../config-node";
+import { buildNode, buildTemplates } from "../tree";
+
+describe("buildNode", () => {
+    const mockTemplatesAccum = {};
+    const mockIgnoreInitialValues: boolean = false;
+
+    it("run with dxKey as props", () => {
+        const mockNode: IConfigNode = {
+            fullName: "",
+            predefinedOptions: {},
+            initialOptions: {
+                center: {
+                    lat: 40.749825,
+                    lng: -73.987963
+                },
+                zoom: 10
+            },
+            options: {
+                controls: true,
+                width: "100%",
+                provider: "google",
+                dxKey: {
+                    google: "X"
+                }
+            },
+            templates: [],
+            configs: {},
+            configCollections: {}
+        };
+        expect(buildNode(mockNode, mockTemplatesAccum, mockIgnoreInitialValues)).toEqual({
+            center: {lat: 40.749825, lng: -73.987963},
+            controls: true,
+            key: {
+                google: "X"
+            },
+            provider: "google",
+            width: "100%",
+            zoom: 10
```

Comment:
A test should be as short as possible. It seems that all other options are not required for this test (same for other tests)

```suggestion
            key: {
                google: "X"
            },
```